docs/docs/building.md

```diff
@@ -4,158 +4,199 @@ title: Building an upgradeable application
 
-## Sync your project with the blockchain with `zos push`
+    contract Basil is Ownable {
+      // Color in RGB.
+      uint256 public red;
+      uint256 public green;
+      uint256 public blue;
 
-This command will deploy your upgradeable application to the blockchain:
-```
-zos push --network <network>
-```
+      uint256 public highestDonation;
 
-The first time you run this command for a specific network, a new `package.zos.<network>.json` will be created. This file will reflect the status of your project in that network.
+      event NewDonation(address indexed donor, uint256 value, uint256 red, uint256 green, uint256 blue);
 
-## Create upgradeability proxies for each of your contracts  
+      function initialize() public isInitializer("Basil", "0") {
+        highestDonation = 10;
+      }
 
-The next commands will deploy new proxies to make your contracts upgradeable:
+      function donate(uint256 _red, uint256 _green, uint256 _blue) public payable {
+        require(_red < 256);
+        require(_green < 256);
+        require(_blue < 256);
+        require(msg.value > highestDonation);
+        red = _red;
+        green = _green;
+        blue = _blue;
+        emit NewDonation(msg.sender, msg.value, red, green, blue);
+      }
+    }
 
-```
-zos create <contract_name_1> --network <network>
-zos create <contract_name_2> --network <network>
-...
-zos create <contract_name_n> --network <network>
-```
+The contract is super simple. If somebody wants to set the light color, they have to make a donation that then goes to cover any plant necessities. If the donation is higher than the previous one, it is accepted, the light color changes and an event is emitted.
 
-Optionally, you can use the `-i` flag to call an initialization/migration function after you create the proxy.
+We set the initial donation amount to 10 wei, and here you will find the only difference to take into account when writting a contract for ZeppelinOS. Before ZeppelinOS, we would have set the initial value using a `constructor` function. In Ethereum, constructors are handled in a very different way compared to normal functions: they are executed during the deployment of the contract to initialize the state variables, and the code of the constructor is never deployed to the blockchain. In ZeppelinOS we rely on proxy contracts that will forward function calls to the contracts with the implementation. A proxy has no access to the constructor to initialize state variables, so instead we use an `initialize` function with the `isInitializer` modifier provided by the `Migratable` contract of `zos-lib`. The modifier receives the name of the contract and a migrationId that we start in 0.
 
-In our simple example, we want to call the `initialize` function, with some value (e.g: 42) thus:
-```
-zos create MyContract -i initialize -p 42 --network development 
-```
+We need to install the `openzeppelin-zos` dependency and to compile the contract:
```

Comment:
the `openzeppelin-zos` dependency -> the `openzeppelin-zos` and `zos-lib` dependencies

Reply:
I still need to check this. But openzeppelin-zos uses Migratable, so it should bring the zos-lib dependency. I found it handy, because then on this whole guide we almost never mention the lib.
Do you want me to add it to be explicit, or because you think it will fail?